client/new-stack-editor/lib/views/flexsplit/index.coffee

```diff
@@ -0,0 +1,89 @@
+kd = require 'kd'
+
+module.exports = class FlexSplit extends kd.View
+
+  @EVENT_EXPAND    = 'FlexSplit.EXPAND'
+  @EVENT_COLLAPSE  = 'FlexSplit.COLLAPSE'
+
+  @EVENT_RESIZED   = 'FlexSplit.RESIZED'
+  @EVENT_EXPANDED  = 'FlexSplit.EXPANDED'
+  @EVENT_COLLAPSED = 'FlexSplit.COLLAPSED'
+
+  @MAX = 100
+  @MIN = 0.0001
+
+  @HORIZONTAL =
+    name      : 'horizontal'
+    axis      : 'y'
+    getter    : 'getHeight'
+
+  @VERTICAL   =
+    name      : 'vertical'
+    axis      : 'x'
+    getter    : 'getWidth'
+
+
+  constructor: (options = {}, data) ->
+
+    options.cssClass   = kd.utils.curry 'flex-split', options.cssClass
+    options.sizes     ?= []
+    options.type      ?= FlexSplit.HORIZONTAL
+    options.resizable ?= yes
+
+    super options, data
+
+    @resizer = null
+    { @type, @name, storage } = @getOptions()
+    @setClass @type.name
+
+    @setupViews()
+
+    storage?.addView this, @name
+
+
+  createResizer: (view, size) ->
+
+    # This needs to be here to prevent circular dependency ~ GG
+    FlexSplitResizer = require './resizer'
+
+    @resizer = @addSubView new FlexSplitResizer { @type, view }
+
+    @forwardEvents @resizer, [
+      FlexSplit.EVENT_EXPANDED
+      FlexSplit.EVENT_RESIZED
+      FlexSplit.EVENT_COLLAPSED
+    ]
+
+    @resizer.on FlexSplit.EVENT_EXPAND, ->
+      if view.parent instanceof FlexSplit
+        view.parent.emit FlexSplit.EVENT_EXPAND
+
+    @resizer.on FlexSplit.EVENT_COLLAPSE, ->
+      if view.parent instanceof FlexSplit
```

Comment:
Can we distinguish with something else? `instanceof` checks are error prone when code is minified Maybe we can generalize this with a field on `kd.Object` like `_type`?

As a side effect, that would remove the need to `require` the file itself in places where we only need this type of check. (it would require you to know the type's string representation, if you don't want to require tho)